client/client-multi/client-c/src/kaa/collections/kaa_list.c

```diff
@@ -309,3 +309,17 @@ void kaa_list_sort(kaa_list_t *list, match_predicate pred)
     }
     list->tail = node;
 }
+
+int32_t kaa_list_hash(kaa_list_t *list, list_node_hash pred)
+{
+    KAA_RETURN_IF_NIL2(list, pred, 0);
+    int32_t result = 1;
+    kaa_list_node_t *node = kaa_list_begin(list);
+    uint64_t element;
+    while (node) {
+        element = pred(node->data);
+        result = 31 * result + (uint32_t) (element ^ (element >> 32));
```

Comment:
Using int32 here will result in UB when it comes to integer overflow. It is easy to hit due to multiplication on every iteration.
See
https://www.securecoding.cert.org/confluence/display/c/INT32-C.+Ensure+that+operations+on+signed+integers+do+not+result+in+overflow

In close-term practice this may force to fail any instrumentation (like -ftrapv flag or runtime-checking tools) that designed to catch integer overflows.